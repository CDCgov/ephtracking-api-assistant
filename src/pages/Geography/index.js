import React, { Component, Fragment } from 'react';

import CodeBlock from '../../components/CodeBlock';
import CIM from '../../components/CIM';
import SelectGeographicType from '../../containers/SelectGeographicType';
import Preview from '../../components/Preview';

class Geography extends Component {
  constructor(props) {
    super(props);
    this.state = {
      measureId: null,
      geographicTypeId: null,
      geographicRollup: false
    };
    this.setMeasureId = this.setMeasureId.bind(this);
    this.setGeographicTypeId = this.setGeographicTypeId.bind(this);
    this.handleInputChange = this.handleInputChange.bind(this);
  }

  setMeasureId(measureId) {
    this.setState({ measureId });
  }

  setGeographicTypeId(geographicTypeId) {
    this.setState({ geographicTypeId });
  }

  handleInputChange(event) {
    const { name, checked } = event.target;
    this.setState({
      [name]: checked
    });
  }

  render() {
    const { measureId, geographicTypeId, geographicRollup } = this.state;
    const validUrl = measureId !== null && geographicTypeId !== null;
    const url = `https://ephtracking.cdc.gov/apigateway/api/v1/geography/${measureId}/${geographicTypeId}/${
      geographicRollup ? 1 : 0
    }`;
    return (
      <Fragment>
        <h1 className="title">
          Retrieving all Geographies for a MeasureId, GeographicType and
          Geographic Rollup
        </h1>
        <h5 className="title is-5">Usage</h5>
        <CodeBlock>
          https://ephtracking.cdc.gov/apigateway/api/{'{'}version{'}'}/geography/{
            '{'
          }measureId{'}'}/{'{'}geographicTypeId{'}'}/{'{'}geographicRollup{'}'}[?apiToken]
        </CodeBlock>
        <hr />
        <h5 className="title is-5">Set parameters</h5>
        <CIM handleSelect={this.setMeasureId} />
        <SelectGeographicType
          handleSelect={this.setGeographicTypeId}
          measureId={measureId}
        />
        <div className="field">
          <label className="checkbox">
            <input
              name="geographicRollup"
              type="checkbox"
              checked={this.state.geographicRollup}
              onChange={this.handleInputChange}
              className="input-checkbox"
            />
            geographicRollup
          </label>
        </div>
        <hr />
        <Preview url={url} validUrl={validUrl} />
      </Fragment>
    );
  }
}

export default Geography;
