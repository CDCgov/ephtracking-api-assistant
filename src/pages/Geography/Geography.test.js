import React from 'react';
import { shallow } from 'enzyme';

import Geography from './';

describe('Page Geography', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Geography />);
    expect(wrapper).toMatchSnapshot();
  });

  it('sets meausreId correctly', () => {
    const wrapper = shallow(<Geography />);
    wrapper.instance().setMeasureId('333');
    expect(wrapper.state()).toMatchObject({ measureId: '333' });
  });

  it('set geographicTypeId correctly', () => {
    const wrapper = shallow(<Geography />);
    wrapper.instance().setGeographicTypeId('1');
    expect(wrapper.state()).toMatchObject({ geographicTypeId: '1' });
  });

  it('handles input geographicRollup check correctly', () => {
    const wrapper = shallow(<Geography />);
    const event = {
      target: {
        type: 'checkbox',
        checked: true,
        name: 'geographicRollup'
      }
    };
    wrapper.instance().handleInputChange(event);
    expect(wrapper.state()).toMatchObject({ geographicRollup: true });
  });
});
