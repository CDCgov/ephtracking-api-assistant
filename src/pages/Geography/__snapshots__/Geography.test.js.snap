// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Geography renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Geography />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <h1
          className="title"
>
          Retrieving all Geographies for a MeasureId, GeographicType and Geographic Rollup
</h1>,
        <h5
          className="title is-5"
>
          Usage
</h5>,
        <CodeBlock>
          https://ephtracking.cdc.gov/apigateway/api/
          {
          version
          }
          /geography/
          {
          measureId
          }
          /
          {
          geographicTypeId
          }
          /
          {
          geographicRollup
          }
          [?apiToken]
</CodeBlock>,
        <hr />,
        <h5
          className="title is-5"
>
          Set parameters
</h5>,
        <CIM
          handleSelect={[Function]}
/>,
        <SelectGeographicType
          handleSelect={[Function]}
          measureId={null}
/>,
        <div
          className="field"
>
          <label
                    className="checkbox"
          >
                    <input
                              checked={false}
                              className="input-checkbox"
                              name="geographicRollup"
                              onChange={[Function]}
                              type="checkbox"
                    />
                    geographicRollup
          </label>
</div>,
        <hr />,
        <Preview
          url="https://ephtracking.cdc.gov/apigateway/api/v1/geography/null/null/0"
          validUrl={false}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Retrieving all Geographies for a MeasureId, GeographicType and Geographic Rollup",
          "className": "title",
        },
        "ref": null,
        "rendered": "Retrieving all Geographies for a MeasureId, GeographicType and Geographic Rollup",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Usage",
          "className": "title is-5",
        },
        "ref": null,
        "rendered": "Usage",
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            "https://ephtracking.cdc.gov/apigateway/api/",
            "{",
            "version",
            "}",
            "/geography/",
            "{",
            "measureId",
            "}",
            "/",
            "{",
            "geographicTypeId",
            "}",
            "/",
            "{",
            "geographicRollup",
            "}",
            "[?apiToken]",
          ],
        },
        "ref": null,
        "rendered": Array [
          "https://ephtracking.cdc.gov/apigateway/api/",
          "{",
          "version",
          "}",
          "/geography/",
          "{",
          "measureId",
          "}",
          "/",
          "{",
          "geographicTypeId",
          "}",
          "/",
          "{",
          "geographicRollup",
          "}",
          "[?apiToken]",
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Set parameters",
          "className": "title is-5",
        },
        "ref": null,
        "rendered": "Set parameters",
        "type": "h5",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "handleSelect": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "handleSelect": [Function],
          "measureId": null,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <label
            className="checkbox"
>
            <input
                        checked={false}
                        className="input-checkbox"
                        name="geographicRollup"
                        onChange={[Function]}
                        type="checkbox"
            />
            geographicRollup
</label>,
          "className": "field",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                checked={false}
                className="input-checkbox"
                name="geographicRollup"
                onChange={[Function]}
                type="checkbox"
/>,
              "geographicRollup",
            ],
            "className": "checkbox",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "checked": false,
                "className": "input-checkbox",
                "name": "geographicRollup",
                "onChange": [Function],
                "type": "checkbox",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            "geographicRollup",
          ],
          "type": "label",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "url": "https://ephtracking.cdc.gov/apigateway/api/v1/geography/null/null/0",
          "validUrl": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <h1
            className="title"
>
            Retrieving all Geographies for a MeasureId, GeographicType and Geographic Rollup
</h1>,
          <h5
            className="title is-5"
>
            Usage
</h5>,
          <CodeBlock>
            https://ephtracking.cdc.gov/apigateway/api/
            {
            version
            }
            /geography/
            {
            measureId
            }
            /
            {
            geographicTypeId
            }
            /
            {
            geographicRollup
            }
            [?apiToken]
</CodeBlock>,
          <hr />,
          <h5
            className="title is-5"
>
            Set parameters
</h5>,
          <CIM
            handleSelect={[Function]}
/>,
          <SelectGeographicType
            handleSelect={[Function]}
            measureId={null}
/>,
          <div
            className="field"
>
            <label
                        className="checkbox"
            >
                        <input
                                    checked={false}
                                    className="input-checkbox"
                                    name="geographicRollup"
                                    onChange={[Function]}
                                    type="checkbox"
                        />
                        geographicRollup
            </label>
</div>,
          <hr />,
          <Preview
            url="https://ephtracking.cdc.gov/apigateway/api/v1/geography/null/null/0"
            validUrl={false}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Retrieving all Geographies for a MeasureId, GeographicType and Geographic Rollup",
            "className": "title",
          },
          "ref": null,
          "rendered": "Retrieving all Geographies for a MeasureId, GeographicType and Geographic Rollup",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Usage",
            "className": "title is-5",
          },
          "ref": null,
          "rendered": "Usage",
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              "https://ephtracking.cdc.gov/apigateway/api/",
              "{",
              "version",
              "}",
              "/geography/",
              "{",
              "measureId",
              "}",
              "/",
              "{",
              "geographicTypeId",
              "}",
              "/",
              "{",
              "geographicRollup",
              "}",
              "[?apiToken]",
            ],
          },
          "ref": null,
          "rendered": Array [
            "https://ephtracking.cdc.gov/apigateway/api/",
            "{",
            "version",
            "}",
            "/geography/",
            "{",
            "measureId",
            "}",
            "/",
            "{",
            "geographicTypeId",
            "}",
            "/",
            "{",
            "geographicRollup",
            "}",
            "[?apiToken]",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Set parameters",
            "className": "title is-5",
          },
          "ref": null,
          "rendered": "Set parameters",
          "type": "h5",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "handleSelect": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "handleSelect": [Function],
            "measureId": null,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <label
              className="checkbox"
>
              <input
                            checked={false}
                            className="input-checkbox"
                            name="geographicRollup"
                            onChange={[Function]}
                            type="checkbox"
              />
              geographicRollup
</label>,
            "className": "field",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  checked={false}
                  className="input-checkbox"
                  name="geographicRollup"
                  onChange={[Function]}
                  type="checkbox"
/>,
                "geographicRollup",
              ],
              "className": "checkbox",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "checked": false,
                  "className": "input-checkbox",
                  "name": "geographicRollup",
                  "onChange": [Function],
                  "type": "checkbox",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              "geographicRollup",
            ],
            "type": "label",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "url": "https://ephtracking.cdc.gov/apigateway/api/v1/geography/null/null/0",
            "validUrl": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
